gh-121999: Change default tarfile filter to 'data' (#122002)
python/cpython: Doc/library/tarfile.rst

   Archives are extracted using a :ref:`filter <tarfile-extraction-filter>`,
   which makes it possible to either limit surprising/dangerous features,
   or to acknowledge that they are expected and the archive is fully trusted.

.. versionchanged:: 3.14

>    The default extraction filter was 'fully trusted' but is now 'data' which

Reviewer: Could be useful to add a reference to the filters here:
```suggestion
   The default extraction filter was :func:`fully_trusted <fully_trusted_filter>` but is now :func:`data <data_filter>` which
```

>    The default extraction filter was :func:`fully_trusted <fully_trusted_filter>` but is now :func:`data <data_filter>` which
>    which disallows dangerous features like links to absolute paths or paths
>    outside the destination.

Reviewer 2: ```suggestion
   Set the default extraction filter to :func:`data <data_filter>`,
   which disallows dangerous features such as links to absolute paths
   or paths outside of the destination. Previously, the filter strategy
   was :func:`fully_trusted <fully_trusted_filter>`.
```

   Archives are extracted using a :ref:`filter <tarfile-extraction-filter>`,
   which makes it possible to either limit surprising/dangerous features,
   or to acknowledge that they are expected and the archive is fully trusted.

.. versionchanged:: 3.14
   Set the default extraction filter to :func:`data <data_filter>`,
   which disallows dangerous features such as links to absolute paths

>    or paths outside of the destination. Previously, the filter strategy

Reviewer 2: Sorry about this one but the `outside of` now feels weird to me :') 

@AA-Turner As a native speaker (you're the only one I know...), should it be "outside the destination", or "outside of"? (or something else entirely?)

Author: I think "outside of" is conventional in North America, less conventional in the UK.
https://learningenglish.voanews.com/a/should-we-think-outside-or-outside-of-the-box-/6434530.html

I think I was reading from PEP 706 when I wrote that update.
> Refuse to extract links (hard or soft) which end up linking to a path outside of the destination. (On systems that don’t support links, tarfile will, in most cases, fall back to creating regular files. This proposal doesn’t change that behaviour.)

   The *filter* argument specifies how ``members`` are modified or rejected
   before extraction.
   See :ref:`tarfile-extraction-filter` for details.
   It is recommended to set this explicitly only if unusual *tar* features
   are required.

   .. warning::

>       The default filter is set to ``filter='data'`` to prevent the most
>       dangerous security issues, read the :ref:`tarfile-extraction-filter`
>       section for details.

Reviewer 2: ```suggestion
      The default filter is set to ``filter='data'`` to prevent the most
      dangerous security issues. Read the :ref:`tarfile-extraction-filter`
      section for details.
```

      See the warning for :meth:`extractall`.

      The default filter is set to ``filter='data'`` to prevent the most

>       dangerous security issues, read the :ref:`tarfile-extraction-filter`

Reviewer 2: ```suggestion
      dangerous security issues. Read the :ref:`tarfile-extraction-filter`
```
